pkg/apis/garden/types.go

```diff
@@ -1202,6 +1202,25 @@ type KubeAPIServerConfig struct {
 	// configuration.
 	// +optional
 	AdmissionPlugins []AdmissionPlugin
+	// AuditConfig contains settings for audit of the api server
+	// +optional
+	AuditConfig *AuditConfig
+}
+
+// AuditConfig contains settings for audit of the api server
+type AuditConfig struct {
+	// AuditPolicy contains audit policy for kube-apiserver
+	// +optional
+	AuditPolicy *AuditPolicy
+}
+
+// AuditPolicy contains audit policy for kube-apiserver
+type AuditPolicy struct {
+	// +optional
+	// AuditPolicy *auditv1beta1.Policy
+
+	// +optional
+	ConfigMapRef *corev1.ObjectReference
```

Comment:
I have talked about this with @vlerenc. I switched from `LocalObjectReference` to `ObjectReference` because of several reasons:
* the Garden operator may define cluster-wide audit policies in some common namespace (`garden` for example). In this way the operator does not have to replicate the audit policies across all namespaces and support multiple replicas of common resource.
* the Garden operator may disallow end users to deploy their audit policies (ConfigMaps) in the namespace of their Project. In this way the end users will be obligated to use only predefined audit policies.

The `ObjectReference` has its own downsides:
* a shoot from project `X` may use audit policy from project `Y`

As a solution, I could limit the `ObjectReference` to refer config maps in the same namespace where the shoot resides, or in the common `garden` namespace.